### ISSUE TITLE
Unloading assets

### ISSUE THREAD

**Issue by Ratysz** [zakarumych/goods#8]
From what I was able to gather, there is currently no way to drop an asset without dropping the whole `Cache` it's in. Is it possible to implement this functionality in some way?

It would enable manual reloading of assets (alleviating somewhat the need for full-on hot reloading) and freeing of resources taken up by assets no longer needed by the application (e.g., unloading old level on level change in a game).

It might be possible to do this by having the cache hold onto weak references to assets, meaning that if there are no `Handle`s to an asset it should be unloaded.

**Comment by zakarumych** [comment]
Yes. This is possible to implement. And must be implemented in some way.
Unloading asset immediately after all Handles to it are dropped is not the best practice. Chances are it will be requested again.

Different scenarios require different solutions.
First of all Cache should be able to unload unused assets on demand. i.e. when application wants to free memory.
For that assets must provide metrics for Cache to decide which should be dropped and which can stay loaded.

Removing specific asset from Cache is possible and seems straightforward to implement.
